go/adbc/driver/snowflake/connection.go

```diff
@@ -73,9 +73,51 @@ type connectionImpl struct {
 	useHighPrecision  bool
 }
 
+func escapeSingleQuoteForLike(arg string) string {
+	if len(arg) == 0 {
+		return arg
+	}
+
+	idx := strings.IndexByte(arg, '\'')
+	if idx == -1 {
+		return arg
+	}
+
+	var b strings.Builder
+	b.Grow(len(arg))
+
+	for {
+		before, after, found := strings.Cut(arg, `'`)
+		b.WriteString(before)
+		if !found {
+			return b.String()
+		}
+
+		if before[len(before)-1] != '\\' {
+			b.WriteByte('\\')
+		}
```

Comment:
I suppose this is to handle pre-escaped characters? But what if the escape is itself escaped? (Or is that not allowed?)

Reply:
I guess not from our own spec :sweat_smile:

Reply:
> I guess not from our own spec 😅, we specify escapes aren't supported at all

Yeah, I pointed this out in #1508

Reply:
Maybe it's time I open a branch for 1.2.0...

Reply:
> Maybe it's time I open a branch for 1.2.0...

I've started some work on that in conjunction with multiple results sets. https://github.com/CurtHagenlocher/arrow-adbc/tree/MoreResults

Reply:
Yea, this is intended to handle pre-escaped characters. The logic is taken from snowflake's JDBC driver, I figured handling one level of escaping was sufficient given our current spec. :smile:

Reply:
> > Maybe it's time I open a branch for 1.2.0...
> 
> I've started some work on that in conjunction with multiple results sets. https://github.com/CurtHagenlocher/arrow-adbc/tree/MoreResults

Oh this is great!